### PR TITLE
Block Editor: Fix React Compiler error for 'BlockProps' util component

### PR DESCRIPTION
**Pull request by Mamaduka**
## What?
Part of #61788.

PR fixes the React Compile error for the 'BlockProps' util component by renaming it during destructure. This is similar to how the commands loader avoids a similar error.

https://github.com/WordPress/gutenberg/blob/8be8e4675637726b0ff3194ccc547307c4a489bc/packages/commands/src/components/command-menu.js#L38

## Testing Instructions
All CI checks should be green; we're just renaming the prop.

### CODE REVIEW

**Review by tyxla — APPROVED** [review]
LGTM 👍 

Thanks @Mamaduka!

File: packages/block-editor/src/hooks/utils.js [hunk 1]
```diff
@@ -562,8 +562,13 @@ export function createBlockEditFilter( features ) {
...
 
-function BlockProps( { index, useBlockProps, setAllWrapperProps, ...props } ) {
-	const wrapperProps = useBlockProps( props );
+function BlockProps( {
+	index,
+	useBlockProps: hook,
+	setAllWrapperProps,
+	...props
+} ) {
+	const wrapperProps = hook( props );
```
Comment (ellatrix) [inline packages/block-editor/src/hooks/utils.js:571]: Why is this not allowed?

Comment (Mamaduka) [reply packages/block-editor/src/hooks/utils.js:571]: Passing hooks as pros and the calling then isn’t allowed. The error in description has link with more info.

Comment (ellatrix) [reply packages/block-editor/src/hooks/utils.js:571]: What if we pass the feature? These are all constant. Or add feature IDs and then grab the feature in the component.

Comment (Mamaduka) [reply packages/block-editor/src/hooks/utils.js:571]: Sorry, I don’t follow. Can you share a code example?

**Pull request merged**